### DISCUSSION TITLE
Cannot figure out how to print \u2019 and other character codes properly

### DISCUSSION THREAD

**Discussion by jjaacckk** [python-escpos/python-escpos#458] [Q&A] [answered]
Hello, 

I am using data from an API and often many symbols (like apostrophes and quotations) are represented by their Unicode character codes. They always print to the terminal fine without issue, but when I try to print to the printer it always encodes them incorrectly and spits out garbage. 

I know that the docs say the input needs to be Unicode and I’m pretty sure \u2019 is a proper Unicode character code.

I am using an Epson TM-T88IIP via USB on MacOS with the latest build of python-ecspos

Thanks in advance,
Jack

**Comment by patkan** [comment, marked as answer]
Hi Jack,
so if you are on the newest (https://pypi.org/project/python-escpos/3.0a8/) version of python-escpos and assuming the library correctly know how to encode those characters, the last option would be that those characters are unknown to your printer.

Some points that you could check:

* Have you set the [profile](https://python-escpos.readthedocs.io/en/latest/api/escpos.html#escpos.escpos.Escpos) correctly?
* Maybe this profile could work: https://mike42.me/escpos-printer-db/#profiles/TM-T88II
* If nothing helps, you could try to search and replace the conflicting characters or use a library such as https://pypi.org/project/Unidecode/

**Reply by jjaacckk** [reply]
Thank you very much!
